### ISSUE TITLE
Refactor `train/mean/loss` to provide a better estimate of the end-of-epoch training loss (?)

### ISSUE THREAD

**Issue by jpdunc23** [ai2cm/ace#486]
The `train/mean/loss` metric is aggregated over all batches in a given epoch and as a result has a different interpretation from `val/mean/loss`. This shouldn't matter much once training has converged, but an important reason to track these two metrics is to diagnose convergence so it would be better to have a `train/mean/loss` metric that is more directly comparable to `val/mean/loss`.

Here are a couple of ideas:
- Give higher weight to batches seen at the end of an epoch (e.g., using EMA in `TrainAggregator`).
- Compute `train/mean/loss` at the end of each epoch using a configurable number of batches.

**Comment by oliverwm1** [comment]
> Compute train/mean/loss at the end of each epoch using a configurable number of batches.

I would prefer this general approach. It could also be configured as a fraction of total training batches, which would make it easier to provide a sensible default and would require less fiddling with this option when changing training datasets.

**Comment by mcgibbon** [comment]
This has been implemented.
